tf2_ros/src/tf2_monitor.cpp

```diff
@@ -202,7 +205,7 @@ class TFMonitor
     if (using_specific_chain_)
     {
       auto tmp = buffer_.lookupTransform(framea_, frameb_, tf2::TimePointZero);
-      double diff = tf2::timeToSec(tf2::get_now()) - tf2_ros::timeToSec(tmp.header.stamp);
+      double diff = tf2_ros::timeToSec(clock_->now()) - tf2_ros::timeToSec(tmp.header.stamp);
```

Comment:
It's ok if the diff is negative. The difference between Time objects is a Duration for which negative values are valid. It would only flow if the delta is larger than the maximum duration, plus or minus in which case it will except and overflow.

Reply:
I have the wrong understanding about what's allowed for a Duration by the sounds of it: could we move the discussion to https://github.com/ros2/rclcpp/issues/525?